netbox/utilities/constants.py

```diff
@@ -93,3 +93,7 @@
     "td": {"align"},
     "th": {"align"},
 }
+
+DATA_SOURCE_SUPPORTED_SOCK_SCHEMAS = ['socks4', 'socks4a', 'socks4h', 'socks5', 'socks5a', 'socks5h']
+DATA_SOURCE_SOCK_RDNS_SCHEMAS = ['socks4h', 'socks4a', 'socks5h', 'socks5a']
+DATA_SOURCE_SUPPORTED_SCHEMAS = ['http', 'https', 'socks4', 'socks4a', 'socks4h', 'socks5', 'socks5a', 'socks5h']
```

Comment:
These aren't relevant to data sources specifically; they're pertinent to the configured HTTP proxies (if any), which are used for many different functions. IMO a name like `HTTP_PROXY_*` would make more sense.